src/optimagic/optimizers/iminuit_migrad.py

```diff
@@ -0,0 +1,102 @@
+from dataclasses import dataclass
+from typing import Callable, Optional
+
+import numpy as np
+from iminuit import Minuit  # type: ignore
+from numpy.typing import NDArray
+
+from optimagic import mark
+from optimagic.optimization.algo_options import (
+    STOPPING_MAXFUN,
+)
+from optimagic.optimization.algorithm import Algorithm, InternalOptimizeResult
+from optimagic.optimization.internal_optimization_problem import (
+    InternalOptimizationProblem,
+)
+from optimagic.typing import AggregationLevel
+
+
+@mark.minimizer(
+    name="iminuit_migrad",
+    solver_type=AggregationLevel.SCALAR,
+    is_available=True,
```

Comment:
Only if `iminuit` is installed. See [here](https://optimagic.readthedocs.io/en/latest/how_to/how_to_add_optimizers.html#write-the-minimal-implementation) for how to handle this.